kubernetes/cosmos/README.md

```diff
@@ -0,0 +1,157 @@
+# Cosmos Helm Chart
+
+Cosmos is a storage backend for Cloudserver that lets you manage data stored on a filesystem and other storage platforms.
+
+## Introduction
+
+This chart bootstraps a Cosmos deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+
+## Prerequisites
+
+Before installing this chart, you must either have a Zenko or a standalone Cloudserver instance running.
+
+## Installing the Chart
+
+1. Configure the cosmos backend as a Cloudserver location constraint.
+
+```sh
+$ cat locationConfig.json
+{
+    "us-east-2": { // This is the region on the rcd.remote configuration
+        "type": "pfs",
+        "objectId": "nfs-42",
+        "legacyAwsBehavior": true,
+        "details": {
+            "bucketName": "nfs",
+            "bucketMatch": true,
+            "serverSideEncryption": true,
+            "supportsVersioning": false,
+            "mountPath": "/data", // This should match the PV path
+            "pfsDaemonEndpoint": { // This should match the cosmos pfsd endpoint
+                "host": "my-release-cosmos-pfsd",
+                "port": "80"
+            }
+        }
+    }
+}
+```
+
+2. Create a PV backed by your desired storage plaform. For example, an NFS-backed PV:
+
+```sh
+$ cat << EOF | kubectl apply -f -
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: nfs-pv
+spec:
+  accessModes:
+  - ReadWriteMany
+  capacity:
+    storage: 300Gi
+  nfs:
+    path: /data
+    server: 10.100.1.42
+  persistentVolumeReclaimPolicy: Retain
+```
+
+3. Create a PVC for the just created PV.
+
+```sh
+$ cat << EOF | kubectl apply -f -
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  annotations:
+    helm.sh/resource-policy: keep
+  name: cosmos
+spec:
+  accessModes:
+  - ReadWriteMany
+  resources:
+    requests:
+      storage: 300Gi
+  volumeName: nfs-pv
+EOF
+```
+
+4. Configure the `rcd.remote` values in the `values.yaml` file. For example:
+
+```yaml
+rcd:
+  remote:
+    accessKey: my-access-key
+    secretKey: my-secret-key
+    endpoint: http://cloudserver.local
+    region: us-east-2
```

Comment:
We will need to expose the ingestion bucket parameter to comply with orbit and other MD ingestion features.